src/rubyfmt.rb

```diff
@@ -2272,6 +2272,96 @@ def format_keyword_with_args(ps, expression, keyword)
+  :string_concat => lambda { |ps, rest| format_string_concat(ps, rest) },
+  :unless => lambda { |ps, rest| format_unless(ps, rest) },
+  :begin => lambda { |ps, rest| format_begin(ps, rest) },
+  :brace_block => lambda { |ps, rest| format_brace_block(ps, rest) },
+  :@float => lambda { |ps, rest| format_float(ps, rest) },
+  :ifop => lambda { |ps, rest| format_ifop(ps, rest) },
+  :hash => lambda { |ps, rest| format_hash(ps, rest) },
+  :aref_field => lambda { |ps, rest| format_aref_field(ps, rest) },
+  :aref => lambda { |ps, rest| format_aref(ps, rest) },
+  :args_add_block => lambda { |ps, rest| format_args_add_block(ps, rest) },
+  :bare_assoc_hash => lambda { |ps, rest| format_bare_assoc_hash(ps, rest) },
+  :defined => lambda { |ps, rest| format_defined(ps, rest) },
+  :until => lambda { |ps, rest| format_until(ps, rest) },
+  :return0 => lambda { |ps, rest| format_return0(ps, rest) },
+  :massign => lambda { |ps, rest| format_massign(ps, rest) },
+  :yield => lambda { |ps, rest| format_yield(ps, rest) },
+  :regexp_literal => lambda { |ps, rest| format_regexp_literal(ps, rest) },
+  :alias => lambda { |ps, rest| format_alias(ps, rest) },
+  :field => lambda { |ps, rest| format_field(ps, rest) },
+  :mrhs_new_from_args => lambda { |ps, rest| format_mrhs_new_from_args(ps, rest) },
+  :dot2 => lambda { |ps, rest| format_dot2(ps, rest) },
+  :dot3 => lambda { |ps, rest| format_dot3(ps, rest) },
+  :yield0 => lambda { |ps, rest| format_yield0(ps, rest) },
+  :@op => lambda { |ps, rest| format_op(ps, rest) },
+  :case => lambda { |ps, rest| format_case(ps, rest) },
+  :@gvar => lambda { |ps, rest| format_gvar(ps, rest) },
+  :sclass => lambda { |ps, rest| format_sclass(ps, rest) },
+  :retry => lambda { |ps, rest| format_empty_kwd(ps, rest, "retry") },
+  :break => lambda { |ps, rest| format_keyword_with_args(ps, rest, "break") },
+  :next => lambda { |ps, rest| format_keyword_with_args(ps, rest, "next") },
+  :while_mod => lambda { |ps, rest| format_while_mod(ps, rest, "while") },
+  :until_mod => lambda { |ps, rest| format_while_mod(ps, rest, "until") },
+  :mlhs => lambda { |ps, rest| format_mlhs(ps, rest) },
+  :dyna_symbol => lambda { |ps, rest| format_dyna_symbol(ps, rest) },
+  :rest_param => lambda { |ps, rest| format_rest_param(ps, rest) },
+  :undef => lambda { |ps, rest| format_undef(ps, rest) },
+  :@cvar => lambda { |ps, rest| format_cvar(ps, rest) },
+  :mlhs_paren => lambda { |ps, rest| format_mlhs_paren(ps, rest) },
+  :mrhs_add_star => lambda { |ps, rest| format_mrhs_add_star(ps, rest) },
+  :while => lambda { |ps, rest| format_while(ps, rest) },
+  :lambda => lambda { |ps, rest| format_lambda(ps, rest) },
+  :rescue_mod => lambda { |ps, rest| format_rescue_mod(ps, rest) },
+  :xstring_literal => lambda { |ps, rest| format_xstring_literal(ps, rest) },
+  :@backref => lambda { |ps, rest| format_backref(ps, rest) },
+  :@CHAR => lambda { |ps, rest| format_character_literal(ps, rest) },
+}
```

Comment:
can you try adding `freeze` here to see if it goes faster still?

Reply:
Added, not a measureable improvement but doesn't hurt.